src/specifier.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as fs from 'fs';
+import {extname, join, relative} from 'path';
+import {promisify} from 'util';
+import * as File from 'vinyl';
+
+import {getFileContents} from './file.js';
+
+const findRoot = require('find-root');
+const readFile = promisify(fs.readFile);
+
+export const ensureDirectoryWithinPath = (path: string) =>
+    (directory: string): boolean => {
+      const relativePath = relative(path, directory);
+      return !!(relativePath && !relativePath.startsWith('.'));
+    };
+
+export const ensureManifestWithinPath = (path: string) => {
+  const testDirectory = ensureDirectoryWithinPath(path);
+  return (directory: string): boolean => {
+    return testDirectory && fs.existsSync(join(directory, 'package.json'));
+  }
+};
+
+export const detectBareSpecifierForPath = async(
+    filePath: string,
+    searchWithinPath: string = process.cwd()): Promise<string> => {
+  const rootPath =
+      findRoot(filePath, ensureManifestWithinPath(searchWithinPath));
+
+  const manifestPath = join(rootPath, 'package.json');
+  const manifest = JSON.parse((await readFile(manifestPath)).toString());
+
+  const {name: packageName} = manifest;
```

Comment:
I'm not familiar with this syntax. It's a destructing statement which also renames it? ie. name from within the manifest object gets renamed to packageName? It's only used once, maybe just use manifest.name.